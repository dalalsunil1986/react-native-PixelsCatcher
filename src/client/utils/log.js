/**
* Copyright (c) Maksym Rusynyk 2018 - present
*
* This source code is licensed under the MIT license found in the
* LICENSE file in the root directory of this source tree.
*/
/* @flow */
import network from './network';

const consoleLog = global.console && global.console.log
  // eslint-disable-next-line no-unused-vars
  ? global.console.log : (...args: any) => {};

const serverLog = async (tag: string, ...args: any) => {
  try {
    network.serverLog({
      tag,
      args,
    });
  } catch (err) {
    if (__DEV__) {
      // eslint-disable-next-line no-console
      console.warn('ERROR:serverLog: ', err.message);
    }
  }
};

const log = {
  v: (tag: string, ...args: any) => {
    consoleLog(tag, ...args);
    serverLog(tag, ...args);
  },

  d: (tag: string, ...args: any) => {
    consoleLog(tag, ...args);
    serverLog(tag, ...args);
  },

  i: (tag: string, ...args: any) => {
    consoleLog(tag, ...args);
    serverLog(tag, ...args);
  },

  w: (tag: string, ...args: any) => {
    consoleLog(`${tag} WARNING:`, ...args);
    serverLog(`${tag} WARNING:`, ...args);
  },

  e: (tag: string, ...args: any) => {
    consoleLog(`${tag} ERROR:`, ...args);
    serverLog(`${tag} ERROR:`, ...args);
  },
};

export default log;
