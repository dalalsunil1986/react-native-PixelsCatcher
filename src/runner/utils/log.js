/* @flow */
const consoleLog = global.console.log;

module.exports = {
  v: (tag: string, ...args: any) => {
    consoleLog(`${tag}:`, ...args);
  },

  d: (tag: string, ...args: any) => {
    consoleLog(`${tag}:`, ...args);
  },

  i: (tag: string, ...args: any) => {
    consoleLog(`${tag}:`, ...args);
  },

  w: (tag: string, ...args: any) => {
    consoleLog(`${tag} WARNING:`, ...args);
  },

  e: (tag: string, ...args: any) => {
    consoleLog(`${tag} ERROR:`, ...args);
  },
};
