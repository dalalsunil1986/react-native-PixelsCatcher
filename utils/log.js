const consoleLog = global.console.log;

module.exports = {
  v: (tag, ...args) => {
    consoleLog(`${tag}:`, ...args);
  },

  d: (tag, ...args) => {
    consoleLog(`${tag}:`, ...args);
  },

  i: (tag, ...args) => {
    consoleLog(`${tag}:`, ...args);
  },

  w: (tag, ...args) => {
    consoleLog(`${tag} WARNING:`, ...args);
  },

  e: (tag, ...args) => {
    consoleLog(`${tag} ERROR:`, ...args);
  },
};
