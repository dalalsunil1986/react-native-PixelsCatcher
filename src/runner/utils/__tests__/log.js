/* @flow */
import log from '../log';

global.console.log = jest.fn();

describe('logging', () => {
  const consoleLog = global.console.log;
  const logAll = () => {
    log.v('v');
    log.d('d');
    log.i('i');
    log.w('w');
    log.e('e');
  };

  afterEach(() => {
    log.setLevel('v');
  });

  it('Default logs i, w and e', () => {
    logAll();
    expect(consoleLog.mock.calls).toMatchSnapshot();
  });

  it('Allows to log including v', () => {
    log.setLevel('v');
    logAll();
    expect(consoleLog.mock.calls).toMatchSnapshot();
  });

  it('Allows to log including d', () => {
    log.setLevel('d');
    logAll();
    expect(consoleLog.mock.calls).toMatchSnapshot();
  });

  it('Allows to log including i', () => {
    log.setLevel('i');
    logAll();
    expect(consoleLog.mock.calls).toMatchSnapshot();
  });

  it('Allows to log including w', () => {
    log.setLevel('w');
    logAll();
    expect(consoleLog.mock.calls).toMatchSnapshot();
  });

  it('Allows to log including e', () => {
    log.setLevel('e');
    logAll();
    expect(consoleLog.mock.calls).toMatchSnapshot();
  });
});
