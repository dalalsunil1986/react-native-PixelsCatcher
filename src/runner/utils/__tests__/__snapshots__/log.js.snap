// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`logging Allows to log including d 1`] = `
Array [
  Array [
    "i:",
  ],
  Array [
    "w WARNING:",
  ],
  Array [
    "e ERROR:",
  ],
  Array [
    "v:",
  ],
  Array [
    "d:",
  ],
  Array [
    "i:",
  ],
  Array [
    "w WARNING:",
  ],
  Array [
    "e ERROR:",
  ],
  Array [
    "d:",
  ],
  Array [
    "i:",
  ],
  Array [
    "w WARNING:",
  ],
  Array [
    "e ERROR:",
  ],
]
`;

exports[`logging Allows to log including e 1`] = `
Array [
  Array [
    "i:",
  ],
  Array [
    "w WARNING:",
  ],
  Array [
    "e ERROR:",
  ],
  Array [
    "v:",
  ],
  Array [
    "d:",
  ],
  Array [
    "i:",
  ],
  Array [
    "w WARNING:",
  ],
  Array [
    "e ERROR:",
  ],
  Array [
    "d:",
  ],
  Array [
    "i:",
  ],
  Array [
    "w WARNING:",
  ],
  Array [
    "e ERROR:",
  ],
  Array [
    "i:",
  ],
  Array [
    "w WARNING:",
  ],
  Array [
    "e ERROR:",
  ],
  Array [
    "w WARNING:",
  ],
  Array [
    "e ERROR:",
  ],
  Array [
    "e ERROR:",
  ],
]
`;

exports[`logging Allows to log including i 1`] = `
Array [
  Array [
    "i:",
  ],
  Array [
    "w WARNING:",
  ],
  Array [
    "e ERROR:",
  ],
  Array [
    "v:",
  ],
  Array [
    "d:",
  ],
  Array [
    "i:",
  ],
  Array [
    "w WARNING:",
  ],
  Array [
    "e ERROR:",
  ],
  Array [
    "d:",
  ],
  Array [
    "i:",
  ],
  Array [
    "w WARNING:",
  ],
  Array [
    "e ERROR:",
  ],
  Array [
    "i:",
  ],
  Array [
    "w WARNING:",
  ],
  Array [
    "e ERROR:",
  ],
]
`;

exports[`logging Allows to log including v 1`] = `
Array [
  Array [
    "i:",
  ],
  Array [
    "w WARNING:",
  ],
  Array [
    "e ERROR:",
  ],
  Array [
    "v:",
  ],
  Array [
    "d:",
  ],
  Array [
    "i:",
  ],
  Array [
    "w WARNING:",
  ],
  Array [
    "e ERROR:",
  ],
]
`;

exports[`logging Allows to log including w 1`] = `
Array [
  Array [
    "i:",
  ],
  Array [
    "w WARNING:",
  ],
  Array [
    "e ERROR:",
  ],
  Array [
    "v:",
  ],
  Array [
    "d:",
  ],
  Array [
    "i:",
  ],
  Array [
    "w WARNING:",
  ],
  Array [
    "e ERROR:",
  ],
  Array [
    "d:",
  ],
  Array [
    "i:",
  ],
  Array [
    "w WARNING:",
  ],
  Array [
    "e ERROR:",
  ],
  Array [
    "i:",
  ],
  Array [
    "w WARNING:",
  ],
  Array [
    "e ERROR:",
  ],
  Array [
    "w WARNING:",
  ],
  Array [
    "e ERROR:",
  ],
]
`;

exports[`logging Default logs i, w and e 1`] = `
Array [
  Array [
    "i:",
  ],
  Array [
    "w WARNING:",
  ],
  Array [
    "e ERROR:",
  ],
]
`;
